src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java

```diff
@@ -142,7 +143,14 @@ private String getQuery(String photoCategory, byte[] uploadedImageBytes)
         }
         return shoppingQuery;
       case "barcode":
-        return "Running shoes";
+        BarcodeImageDetector barcodeImageDetector = new BarcodeImageDetector();
+        String barcodeQuery;
+        try {
+          barcodeQuery = barcodeImageDetector.detect(uploadedImageBytes);
+        } catch (PhotoDetectionException exception) {
+          throw exception;
+        }
+        return barcodeQuery;
```

Comment:
Is there a benefit of catching an exception and then just throwing it again?
Wouldn't `return barcodeImageDetector.detect(uploadedImageBytes);` do the same?

Reply:
Indeed, solved!